napari/_app_model/context/_layerlist_context.py

```diff
@@ -130,6 +130,10 @@ def _active_is_image_3d(s: LayerSel) -> bool:
     )
 
 
+def _empty_shapes_layer_selected(s: LayerSel) -> bool:
+    return any(x._type_string == "shapes" and len(x.data) == 0 for x in s)
```

Comment:
```suggestion
    return any(x._type_string == "shapes" and not x.size for x in s)
```

as data are numpy arrays then checking size is better than len